integration/deltaspike-data/api/src/main/java/com/blazebit/persistence/deltaspike/data/ExtendedQueryResult.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2014 - 2018 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.deltaspike.data;
+
+import com.blazebit.persistence.PagedList;
+import org.apache.deltaspike.data.api.QueryResult;
+
+import javax.persistence.FlushModeType;
+import javax.persistence.LockModeType;
+import javax.persistence.metamodel.SingularAttribute;
+import java.util.List;
+
+/**
+ * An extended version of {@link QueryResult}.
+ *
+ * @param <E> Entity type.
+ *
+ * @author Christian Beikov
+ * @since 1.2.0
+ */
+public interface ExtendedQueryResult<E> extends QueryResult<E> {
+
+    /**
+     * Returns the result list of the page for which this query was constructed for without executing a count query.
+     *
+     * @return The result list of the requested page
+     */
+    List<E> getPageResultList();
+
+    /**
+     * Returns a {@link PagedList} if paginated containing the result list of the requested page and optionally the total count depending on {@link #withCountQuery(boolean)}
```

Comment:
If paginated, returns a {@link PagedList} containing the result list of the requested page and optionally the total count depending on {@link #withCountQuery(boolean)}. Otherwise, returns a {@link List}.